python/dlr/tf_model.py

```diff
@@ -0,0 +1,143 @@
+import tensorflow as tf
+from tensorflow.python.tools import saved_model_utils
+
+from .api import IDLRModel
+from .neologger import create_logger
+from .metadata import VERSION
+
+class TFModelImpl(IDLRModel):
+    """
+    TFModelImpl is a wrapper on top of tensorflow which implements DLRModel API
+    Parameters
+    ----------
+    model_path : str
+        Full path to the directory containing the saved model
+    dev_type : str
+        Device type ('cpu' or 'gpu')
+    dev_id : int
+        Device ID
+    """
+
+    def __init__(self, model_path, dev_type=None, dev_id=None, error_log_file=None, use_default_dlr=False):
+
+        self.model_path = model_path
+        self.logger = create_logger(log_file=error_log_file)
+        if dev_type is not None:
+            devices = ["cpu", "gpu"]
+            if dev_type not in devices:
+                raise ValueError(
+                    "Invalid device type {}. Valid devices: {}".format(dev_type, devices))
+        if dev_type == 'gpu':
```

Comment:
We do not ask TF to run the saved_model on a particular Device.  Looks like `dev_type` and `dev_id` params are meaningless for TF2 adapter. 
It should be ok for Jetson users to skip `dev_type` and `dev_id` params. Example:
```
model=dlr.DLRModel("./saved_tf_trt_model")
```
Lets `list_physical_devices('GPU')` and `set_memory_growth(True)` for all of them regardless of `dev_type` and `dev_id`.

2. If user tries to set `dev_type` and `dev_id` we should show warn message saying that `dev_type` and `dev_id` are not supported for TF2 Models and the params are ignores.

Reply:
done